### ISSUE TITLE
This library seems to use static methods, multiple main JFrames

### ISSUE THREAD

**Issue by KlemenDEV** [andrewauclair/ModernDocking#101]
So in order to initialize the docking system, we call

`DockingUI.initialize(this);`

How would I go with an app with multiple projects open, each with its own JFrame and set of docking?

Can this method be called on multiple frames?

**Comment by andrewauclair** [comment]
That is currently not possible. There is a single instance of Docking per JVM.

The framework probably should have the flexibility for this. The static methods are far more convenient for what I designed it for.

**Comment by andrewauclair** [comment]
@KlemenDEV I have this roughly implemented with a new `DockingInstance` class that has the same interface (but not static) as `Docking`.

I'm currently looking into how I can better separate the static and non-static setups. I would like to keep the static stuff for convenience for other use cases where that's all that is required. But while setting up some tests for the new multi-config I noticed it was really easy to accidentally call one of the static methods somewhere and cause issues.

**Comment by KlemenDEV** [comment]
Thanks, I will take a look.

I think your approach is a good idea with keeping easy system for single JFrame apps, while also offering more extensive system for more complex systems.

**Comment by andrewauclair** [comment]
I'm going to experiment with changing the packing to the following:

```
docking-api
    docking-ui
    docking-single-app
    docking-multi-app
```

Trying this out locally to see how well I can keep single and multi separated.

**Comment by andrewauclair** [comment]
Just published 0.9.0 which will have this feature. Maven info for the package is:

```
groupId:     io.github.andrewauclair
    artifactId:  modern-docking-multi-app
    version:     0.9.0
```
